lambdas/corpus-scheduler-lambda/src/graphQlApiCalls.ts

```diff
@@ -1,28 +1,72 @@
-import config from './config';
 import fetch from 'node-fetch';
+
 import {
   CreateApprovedCorpusItemApiInput,
   CreateScheduledItemInput,
   UrlMetadata,
 } from 'content-common';
 import {
   ApprovedCorpusItemOutput,
+  getApprovedCorpusItemByUrl as getApprovedCorpusItemByUrlCommon,
+  getUrlMetadata as getUrlMetadataCommon,
+  GraphQlApiCallHeaders,
+} from 'lambda-common';
+
+import {
   ApprovedCorpusItemWithScheduleHistoryOutput,
   ScheduledCorpusItemWithApprovedCorpusItemOutput,
 } from './types';
 
 export const sleep = async (ms: number) => {
   await new Promise((resolve) => setTimeout(resolve, ms));
 };
+
+/**
+ * this function wraps the common function for the purposes of easily mocking in tests
+ *
+ * @param adminApiEndpoint string
+ * @param graphHeaders GraphQlApiHeaders object
+ * @param url string
+ * @returns ApprovedCorpusItemOutput
+ */
+export async function getApprovedCorpusItemByUrl(
+  adminApiEndpoint: string,
+  graphHeaders: GraphQlApiCallHeaders,
+  url: string,
+): Promise<ApprovedCorpusItemOutput | null> {
+  return await getApprovedCorpusItemByUrlCommon(
+    adminApiEndpoint,
+    graphHeaders,
+    url,
+  );
+}
```

Comment:
I've sometimes used wrapper functions to be able to write tests against code that was hard to patch. The downsides of wrapper functions are that they add boilerplate code, and that the wrapper itself is usually not covered. Tests still pass for me when the implementation is changed to this:
```typescript
  return await getApprovedCorpusItemByUrlCommon(
    url,
    graphHeaders,
    adminApiEndpoint,
  );
```

In what way are Jest spies are easier to use? Is it some issue with msw that we could address?

Reply:
as for jest, it's lighter weight. with msw, we are initiating a `fetch` call and then intercepting that by URL, and mocking the returned data. jest skips the `fetch` call and interception entirely and just returns that same mocked data.